kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/transaction/TransactionCoordinator.java

```diff
@@ -230,48 +232,70 @@ public void handleInitProducerId(String transactionalId,
             CompletableFuture<ErrorsAndData<Optional<CoordinatorEpochAndTxnMetadata>>>
                     epochAndTxnMetaFuture = new CompletableFuture<>();
             if (!existMeta.getData().isPresent()) {
-                producerIdManager.generateProducerId().whenComplete((pid, throwable) -> {
-                    short producerEpoch = 0;
-                    if (throwable != null) {
-                        responseCallback.accept(new InitProducerIdResult(
-                                -1L,
-                                producerEpoch
-                                , Errors.UNKNOWN_SERVER_ERROR));
-                        return;
-                    }
-                    TransactionMetadata newMetadata = TransactionMetadata.builder()
-                            .transactionalId(transactionalId)
-                            .producerId(pid)
-                            .producerEpoch(producerEpoch)
-                            .state(TransactionState.EMPTY)
-                            .topicPartitions(new HashSet<>())
-                            .txnLastUpdateTimestamp(time.milliseconds())
-                            .build();
-                    epochAndTxnMetaFuture.complete(txnManager.putTransactionStateIfNotExists(newMetadata));
-                });
+                if (existMeta.hasErrors()) {
```

Comment:
If the existing metadata is present and has errors, maybe we need to remove the existing meta from the cache? Normally, if the metadata of one transaction is present, it shouldn't have errors.

Reply:
In this case metadata don't exists and has errors, we should complete the future by follow the kafka transaction behavior.

Reply:
Ok, get it.